### PR TITLE
[WIP][core] Dummy character system

### PR DESCRIPTION
**Pull request by zach2good**
image
image

**Concept**
- Creepy mannequin heads, random faces, random sex/races.
- Player characters, with their packet queues blocked off, with random pathfinding controllers.
- Generate packets for _real_ players to observe and measure improvements to packet system.
- Kinda nice to see lots of player-ish things running around in city zones

**_I affirm:_**
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] that I agree to LandSandBoat's [Limited Contributor License Agreement](https://github.com/LandSandBoat/server/blob/base/.github/CONTRIBUTOR_AGREEMENT.md), as written on this date
- [x] that I have **read the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)**
- [x] that I've _**tested my code and things my code changed**_ since the last commit in the PR, and will test after any later commits

### CODE REVIEW

File: src/map/entities/charentity.h [hunk 1]
```diff
@@ -310,15 +310,16 @@ class CCharEntity : public CBattleEntity
...
-    void          erasePackets(uint8 num); // erase num elements from front of packet list
-    virtual void  HandleErrorMessage(std::unique_ptr<CBasicPacket>&) override;
+    virtual void clearPacketList();                         // отчистка PacketList
+    virtual void pushPacket(CBasicPacket*);                 // добавление копии пакета в PacketList
+    virtual void pushPacket(std::unique_ptr<CBasicPacket>); // push packet to packet list
+    virtual bool isPacketListEmpty();                       // проверка размера PacketList
+    virtual auto popPacket() -> CBasicPacket*;              // получение первого пакета из PacketList
+    virtual auto getPacketList() -> PacketList_t;           // returns a COPY of packet list
+    virtual auto getPacketCount() -> size_t;
+    virtual void erasePackets(uint8 num); // erase num elements from front of packet list
```
Comment (zach2good) [inline src/map/entities/charentity.h:320]: TODO: What is the performance penalty here, probably pretty bad...

Comment (zach2good) [reply src/map/entities/charentity.h:320]: Could we store all of these in a separate object? ie: `CCharEntity` has a member: `std::unique_ptr<CPacketListContainer>` which we populate on `CCharEntity contruction`, but we set as nullptr for dummies. Think pImp pattern?

**Comment by zach2good** [comment]
Don't think I ever want this to make it into the actual codebase. Will save the patch and write an article about it

**Comment by zach2good** [comment]
Article, linked from the development wiki page:
https://github.com/LandSandBoat/server/wiki/Dummy-System

**Comment by zanglikun** [comment]
it will be merg to master banch?

**Comment by zach2good** [comment]
No, this was just a proof of concept, it'll never be merged into master/base

**Pull request closed without merging**